larray/viewer/model.py

```diff
@@ -190,30 +186,33 @@ def get_data(self):
         """Return data"""
         return self._data
 
-    def set_data(self, data, xlabels=None, ylabels=None, changes=None,
-                 bg_gradient=None, bg_value=None):
-        self._set_data(data, xlabels, ylabels, changes, bg_gradient, bg_value)
+    def set_data(self, data, changes=None, bg_gradient=None, bg_value=None):
+        self._set_data(data, changes, bg_gradient, bg_value)
         self.reset()
 
-    def _set_data(self, data, xlabels, ylabels, changes=None, bg_gradient=None, bg_value=None):
+    def _set_data(self, data, changes=None, bg_gradient=None, bg_value=None):
         if changes is None:
             changes = {}
         if data is None:
-            data = np.empty(0, dtype=np.int8).reshape(0, 0)
-        if data.dtype.names is None:
-            dtn = data.dtype.name
+            data = np.empty((0, 0), dtype=np.int8)
+
+        # convert input data to LArray if not
+        if not isinstance(data, (np.ndarray, pd.DataFrame, la.LArray)):
```

Comment:
Can't we just let aslarray handle this (ie remove the explicit test)? just say that we accept anything which can be converted to an larray (basically anything I think)?